From 9d74b1f619507ad136af207ec9510e5fbef7f933 Mon Sep 17 00:00:00 2001
From: Samreen Nilofer <samreen.nilofer@intel.com>
Date: Tue, 23 Jun 2015 16:33:33 +0530
Subject: [PATCH] ASoC: Intel: T4/HR: Add the be_fixup function

The realtek codec's hw_param doesn't cache the clock details
instead it uses it during hw_param itself. Hence the need
to configure the clock settings before the codec hw_params
are invoked.

This is applicable for hostless stream utilizing the codec,
(i.e: aware and wake on voice usecases)

Change-Id: I030120e3f4f1ec9e6e29a5f2c05874f947bb9211
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-12531
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/intel/board/cht_bl_dpcm_rt5672.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
index 360d5e89f2de..c74708b2badc 100644
--- a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
+++ b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
@@ -529,7 +529,8 @@ static int cht_set_slot_fmt_clk(struct snd_soc_dai *codec_dai,
 	unsigned int fmt;
 	int ret;
 
-	pr_debug("Enter:%s", __func__);
+	pr_debug("Enter:%s dai name %s rate %d\n", __func__,
+			codec_dai->name, rate);
 
 	/* proceed only if dai is valid */
 	if (strncmp(codec_dai->name, "rt5670-aif1", 11))
@@ -579,7 +580,7 @@ static int cht_aif1_hw_params(struct snd_pcm_substream *substream,
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct snd_soc_dai *codec_dai = rtd->codec_dai;
 
-	pr_debug("Enter:%s", __func__);
+	pr_debug("Enter:%s\n", __func__);
 	return cht_set_slot_fmt_clk(codec_dai, params_rate(params));
 }
 
@@ -596,6 +597,13 @@ static const struct snd_soc_pcm_stream cht_dai_params = {
 	.channels_max = 2,
 };
 
+static int cht_codec_loop_fixup(struct snd_soc_dai_link *dai_link,
+				struct snd_soc_dai *dai)
+{
+	pr_debug("Enter:%s\n", __func__);
+	return cht_set_slot_fmt_clk(dai, dai_link->params->rate_max);
+}
+
 static int cht_codec_fixup(struct snd_soc_pcm_runtime *rtd,
 			    struct snd_pcm_hw_params *params)
 {
@@ -866,6 +874,7 @@ static struct snd_soc_dai_link cht_dailink[] = {
 		.dai_fmt = SND_SOC_DAIFMT_DSP_B | SND_SOC_DAIFMT_IB_NF
 			| SND_SOC_DAIFMT_CBS_CFS,
 		.params = &cht_dai_params,
+		.be_fixup = cht_codec_loop_fixup,
 		.dsp_loopback = true,
 	},
 	{
-- 
1.9.1

